maven-model-builder/src/main/java/org/apache/maven/model/validation/DefaultModelValidator.java

```diff
@@ -829,14 +825,33 @@ private boolean validateId( String fieldName, ModelProblemCollector problems, Se
         }
         else
         {
-            boolean match = ID_REGEX.matcher( id ).matches();
-            if ( !match )
+            if ( !isValidId( id ) )
             {
                 addViolation( problems, severity, version, fieldName, sourceHint,
                               "with value '" + id + "' does not match a valid id pattern.", tracker );
+                return false;
+            }
+            return true;
+        }
+    }
+
+    private boolean isValidId( String id )
```

Comment:
Should this be `static`? (in order to prevent accidental access to instance fields)